### ISSUE TITLE
Confusing program output

### ISSUE THREAD

**Issue by DanRStevens** [OutpostUniverse/OP2MapImager#12]
I tried running the program on Linux and saw the following output:
```
~/code/opu/opu-git-svn/GameDownload/Outpost2/trunk$ ~/code/opu/OP2MapImager/.build/bin/op2mapimager cm03.map
Render initialized (May take up to 45 seconds): cm03.map
vector::_M_default_append
Run without arguments to see usage message.
```

I expected this would match the example usage:
```
 * OP2MapImager mapFilename.[map|OP2]
```

It appears the command failed, as no image output was produced. Additionally, the "vector::_M_default_append" looks like a debug message of sorts.

For testing purposes, I had extracted "cm03.map" from the "maps.vol" archive (and only that one file).

**Comment by Brett208** [comment]
I haven't been able to reproduce this yet. I successfully rendered a map that was already extracted. OP2MapImager failed gracefully when unable to find the tilesets. It also managed to extract the tilesets from the archive and render properly. When presented with the older format tilesets, it failed gracefully as well.

So, I don't know if this is a Linux specific error, or if I'm not testing correctly. But yes, you should be able to run the command cm03.map and render it properly. So something is breaking unexpectedly.

**Comment by DanRStevens** [comment]
Small update here which might help.

I applied the fix in PR #14, and re-ran the command from Issue #11. I had also deleted the extracted map file, so there were no files extracted from archives in my Outpost 2 folder (no maps, no tilesets). The output is now closer to what I saw in this issue, including the strange vector message. Of note, the strange vector message appeared once, after all output detailing the attempts to process the packed files:
```
~/code/opu/OP2MapImager$ .build/bin/op2mapimager ../opu-git-svn/GameDownload/Outpost2/trunk/
File count found for rendering: 91
--------------------------------------------------

Render initialized (May take up to 45 seconds): Ashes.map
Unable to locate Ashes.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): bhsurv.map
Unable to locate bhsurv.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): cm01.map
Unable to locate cm01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): cm02.map
Unable to locate cm02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): cm03.map
Unable to locate cm03.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): cm04.map
Unable to locate cm04.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): Crossfire.map
Unable to locate Crossfire.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): demo01.map
Unable to locate demo01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): demo02.map
Unable to locate demo02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden01.map
Unable to locate eden01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden02.map
Unable to locate eden02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden03.map
Unable to locate eden03.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden04.map
Unable to locate eden04.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden05.map
Unable to locate eden05.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden06.map
Unable to locate eden06.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden07.map
Unable to locate eden07.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden08.map
Unable to locate eden08.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden09.map
Unable to locate eden09.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden10.map
Unable to locate eden10.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden11.map
Unable to locate eden11.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): eden12.map
Unable to locate eden12.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): edendemo.map
Unable to locate edendemo.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): fl01.map
Unable to locate fl01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): fraxsurv.map
Unable to locate fraxsurv.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_01.map
Unable to locate mp2_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_02.map
Unable to locate mp2_02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_03.map
Unable to locate mp2_03.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_05.map
Unable to locate mp2_05.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_07.map
Unable to locate mp2_07.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_08.map
Unable to locate mp2_08.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_10.map
Unable to locate mp2_10.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_11.map
Unable to locate mp2_11.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp2_12.map
Unable to locate mp2_12.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp3_01.map
Unable to locate mp3_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp3_02.map
Unable to locate mp3_02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp3_08.map
Unable to locate mp3_08.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_01.map
Unable to locate mp4_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_03.map
Unable to locate mp4_03.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_04.map
Unable to locate mp4_04.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_05.map
Unable to locate mp4_05.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_06.map
Unable to locate mp4_06.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_07.map
Unable to locate mp4_07.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_08.map
Unable to locate mp4_08.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_09.map
Unable to locate mp4_09.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_10.map
Unable to locate mp4_10.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_11.map
Unable to locate mp4_11.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp4_12.map
Unable to locate mp4_12.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp6_01.map
Unable to locate mp6_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mp6_03.map
Unable to locate mp6_03.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mpdemo.map
Unable to locate mpdemo.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): mpdemohax.map
Unable to locate mpdemohax.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): newworld.map
Unable to locate newworld.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on1_01.map
Unable to locate on1_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on2_01.map
Unable to locate on2_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on2_02.map
Unable to locate on2_02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on4_01.map
Unable to locate on4_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on4_02.map
Unable to locate on4_02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on4_03.map
Unable to locate on4_03.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on6_01.map
Unable to locate on6_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on6_03.map
Unable to locate on6_03.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): on6_04.map
Unable to locate on6_04.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): OnTheRun.map
Unable to locate OnTheRun.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_01.map
Unable to locate opu_01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_02.map
Unable to locate opu_02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_09.map
Unable to locate opu_09.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_10.map
Unable to locate opu_10.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_11.map
Unable to locate opu_11.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_12.map
Unable to locate opu_12.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_15.map
Unable to locate opu_15.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_z1.map
Unable to locate opu_z1.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_z2.map
Unable to locate opu_z2.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): opu_z3.map
Unable to locate opu_z3.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): outcaster.map
Unable to locate outcaster.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply01.map
Unable to locate ply01.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply02.map
Unable to locate ply02.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply03.map
Unable to locate ply03.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply05.map
Unable to locate ply05.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply06.map
Unable to locate ply06.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply07.map
Unable to locate ply07.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply08.map
Unable to locate ply08.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply09.map
Unable to locate ply09.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply11.map
Unable to locate ply11.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): ply12.map
Unable to locate ply12.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): plydemo.map
Unable to locate plydemo.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): RescueEscort.map
Unable to locate RescueEscort.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): SandyValley.map
Unable to locate SandyValley.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): survatw.map
Unable to locate survatw.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): survaxen.map
Unable to locate survaxen.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): tutorial.map
Unable to locate tutorial.map within directory or within an archive (vol or clm) in the directory.

Render initialized (May take up to 45 seconds): SGAME6.OP2
vector::_M_default_append
Run without arguments to see usage message.
```

**Comment by DanRStevens** [comment]
I tested this again with the current version. I no longer see the message:

```
vector::_M_default_append
Run without arguments to see usage message.
```

----

I updated the command run in the first post and got the following:
```
~/code/opu/opu-git-svn/GameDownload/Outpost2/trunk$ ~/code/opu/OP2MapImager/OP2MapImager cm03.map
Render initialized (May take up to 45 seconds): cm03.map
Could not open file: cm03.map
```

For ease of testing, I updated this command to run from the project folder:
```
~/code/opu/OP2MapImager$ ./OP2MapImager ../opu-git-svn/GameDownload/Outpost2/trunk/cm03.map
Render initialized (May take up to 45 seconds): cm03.map
Unable to locate ../opu-git-svn/GameDownload/Outpost2/trunk/cm03.map within directory or within an archive (vol or clm) in the directory.
```

The output however is not the same, so a different code path is being used.

I also tried testing a command corresponding to my later post:
```
~/code/opu/OP2MapImager$ ./OP2MapImager ../opu-git-svn/GameDownload/Outpost2/trunk/
File count found for rendering: 89
--------------------------------------------------

Render initialized (May take up to 45 seconds): Ashes.map
Unable to locate Ashes.map within directory or within an archive (vol or clm) in the directory.

...

Render initialized (May take up to 45 seconds): tutorial.map
Unable to locate tutorial.map within directory or within an archive (vol or clm) in the directory.

Renders Complete!
--------------------------------------------------
```

It still fails for packed map files, however there is no longer confusing error output.

----

It may be worth considering if we can image files from within archives. That's not really this current issue though.

Perhaps this issue should be closed now.
